### ISSUE TITLE
pages/_document.js throwing no-document-import-in-page

### ISSUE THREAD

**Issue by markxoe** [vercel/next.js#28971]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

16.7.0

### What browser are you using?

Firefox

### What operating system are you using?

Windows 10

### How are you deploying your application?

Only Development

### Describe the Bug

I am getting the @next/next/no-document-import-in-page linting error, but I am importing the document module in the `pages/_document.js` file

Looks funny when displayed:
```
./pages/_document.js
1:1  Error: next/document should not be imported outside of pages/_document.js. See https://nextjs.org/docs/messages/no-document-import-in-page.  @next/next/no-document-import-in-page
```

### Expected Behavior

Not getting an linting error

### To Reproduce

Use a Windows machine, create an Next.js App and create a `pages/_document.js` file according to the docs (https://nextjs.org/docs/advanced-features/custom-document) and lint using `npm run lint`

**Comment by ristomatti** [comment]
We're also seeing this with developers on Windows. There's a PR that should fix this already merged https://github.com/vercel/next.js/pull/28745, it's part of [v11.1.3-canary.4] release (https://github.com/vercel/next.js/releases/tag/v11.1.3-canary.4).

Curiously, explicitly installing `@next/eslint-plugin-next@11.1.3-canary.7`fixed this for us while we were on v11.0.1 but now broke again when upgrading to v11.1.2, even if explicitly installing the above package.

**Comment by andrei9669** [comment]
https://github.com/vercel/next.js/issues/28596

**Comment by housseindjirdeh** [comment]
@markxoe Which version of `eslint-config-next` (or `@next/eslint-plugin-next`) are you using? If you're using an older version than `11.1.3-canary.7`, could you:

1. Upgrade to `11.1.3-canary.7` or a later canary version
2. Run `npm run lint --no-cache` to clear the cache

Let me know if you're still seeing this issue after these steps!

**Comment by markxoe** [comment]
Upgrading eslint-plugin-next did the job 👍

**Comment by ristomatti** [comment]
@housseindjirdeh If I remember correctly, this is what I asked one of the Windows devs to try but can't be sure. We decided to disable the rule for now and see if it gets fixed with the final 11.1.3 (or newer) release. There's a chance this could have been a side effect of npm version mismatch between the devs we noticed a day later

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
